session-replacement/src/main/java/com/amadeus/session/repository/redis/RedisSessionRepository.java

```diff
@@ -530,4 +541,18 @@ public void sessionIdChange(SessionData sessionData) {
     redis.publish(redirectionsChannel, encode(sessionData.getOldSessionId() + ':' + sessionData.getId()));
     expirationManager.sessionIdChange(sessionData);
   }
+
+  @Override
+  public boolean isConnected() {
+    try {
+      redis.info("server");
+      return true;
+    }
+    catch ( Exception e ) {
```

Comment:
Are there no other ways to detect that Redis is disconnected than by catching an exception?

I would rather we find another way. If not possible, then can we at least be more specific on the Exception caught?